spec/main_spec.rb

```diff
@@ -0,0 +1,55 @@
+require '../lib/board.rb'
+require '../lib/player.rb'
+describe Board do
+  board = Board.new
+  describe '#get_color' do
+    it "returns 'O' if 'X' is passed as argument" do
+      expect(board.get_color('X')).to eql('O')
+    end
+    it "returns 'X' if 'O' is passed as argument" do
+      expect(board.get_color('O')).to eql('X')
+    end
+    it 'returns error if no argument is given' do
+      expect { board.get_color }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#valid_move?' do
+    it 'returns true if a move is valid' do
+      expect(board.valid_move?(1)).to be true
+    end
+    it 'returns false if a move is not valid' do
+      expect(board.valid_move?(10)).to be false
+    end
+    it 'returns error if no argument is given' do
+      expect { board.valid_move? }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#display_grid' do
+    it 'returns the grid of string if correct position is passed' do
+      expect(board.display_grid(3, 'X')).to be_instance_of(String)
+    end
+    it 'returns error if invalid argument is paased' do
+      expect { board.display_grid(10, 'X') }.to raise_error(TypeError)
+    end
+  end
+end
+
+describe Player do
```

Comment:
[OPTIONAL]
Consider having `Board` and `Player` tests in separate files like `board_spec.rb` and `player_spec.rb` for separation of concerns.

But seeing as these tests are not lengthy, this is fine. But it's good practice to separate. 🙂